src/NLog/Targets/Target.cs

```diff
@@ -722,6 +724,39 @@ protected string RenderLogEvent(Layout layout, LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Renders the event info in layout.
+        /// </summary>
+        /// <param name="layout">The layout.</param>
+        /// <param name="logEvent">The event info.</param>
+        /// <param name="defaultValue">The default value when <paramref name="layout"/> and/or <paramref name="logEvent"/> is <c>null</c></param>
+        /// <returns>Value representing log event.</returns>
+        protected T RenderLogEvent<T>([CanBeNull] Layout<T> layout, [CanBeNull] LogEventInfo logEvent, T defaultValue = default(T))
+        {
+            if (layout == null || logEvent == null)
+                return defaultValue;
+
+            if (OptimizeBufferReuse)
```

Comment:
Think that `OptimizeBufferReuse` should be marked obsolete, and it should no longer be used anywhere.

Reply:
I don't know what to do here now :|

Reply:
I would stop checking the ``OptimizeBufferReuse`-property. And in a new PR change the property into being obsolete and by default always return true.

Reply:
But why is that needed in this PR? Do I use OptimizeBufferReuse wrong here?

Reply:
It is just because you are creating new code, that depends on a member-variable that is about to be marked as obsolete (and will default to always returning `true`). So just remove remove the `if`-statement.

Reply:
As double check, with "remove the if-statement." - you mean remove the if condition and keep the body, isn't? (so essentially we will remove the else branch)